utils/src/main/java/org/apache/cloudstack/utils/security/DigestHelper.java

```diff
@@ -93,4 +95,33 @@ public static boolean isAlgorithmSupported(String checksum) {
         }
         return true;
     }
+
+    /**
+     * Checks if the algorithm prefix is provided on the checksum
+     */
+    public static void checksumSanity(String checksum) {
```

Comment:
what about a method name with an action that can describe what the method is doing. E.G. `validateChecksumString`, `checkSanityOfTemplateHashes`, and so on.